### ISSUE TITLE
after convert model from .pt to .engine, WHY the gpu memory usage even more?

### ISSUE THREAD

**Issue by huizhilei** [ultralytics/yolov5#8555]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

when i use best.pt to predict, the gpu memory usage is 1391M
but after convert model from .pt to .engine, and use best.engine to predict ,the the gpu memory usage is 1455M
I already use --half  parameter。

### Additional

root@8fad79c545c1:~/yolov5# python export.py --include engine --device 0 --half
export: data=data/mydata.yaml, weights=/root/yolov5/runs/train/exp4/weights/best.pt, imgsz=[640, 640], batch_size=1, device=0, half=True, inplace=False, train=False, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=12, verbose=False, workspace=4, nms=False, agnostic_nms=False, topk_per_class=100, topk_all=100, iou_thres=0.45, conf_thres=0.25, include=['engine']
YOLOv5 🚀 v6.1-253-g75bbaa8 Python-3.6.9 torch-1.11.0.dev20211227+cu111 CUDA:0 (Tesla T4, 15110MiB)

Fusing layers... 
Model summary: 213 layers, 7031701 parameters, 0 gradients, 15.8 GFLOPs

PyTorch: starting from /root/yolov5/runs/train/exp4/weights/best.pt with output shape (1, 25200, 13) (13.8 MB)
Python 3.7.0 required by YOLOv5, but Python 3.6.9 is currently installed

ONNX: starting export with onnx 1.11.0...
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
WARNING: The shape inference of prim::Constant type is missing, so it may result in wrong shape inference for the exported graph. Please consider adding it in symbolic function.
ONNX: export success, saved as /root/yolov5/runs/train/exp4/weights/best.onnx (13.6 MB)

TensorRT: starting export with TensorRT 8.4.1.5...
[07/12/2022-09:29:29] [TRT] [I] [MemUsageChange] Init CUDA: CPU +293, GPU +0, now: CPU 3102, GPU 3220 (MiB)
[07/12/2022-09:29:30] [TRT] [I] [MemUsageChange] Init builder kernel library: CPU +208, GPU +68, now: CPU 3328, GPU 3288 (MiB)
[07/12/2022-09:29:30] [TRT] [I] ----------------------------------------------------------------
[07/12/2022-09:29:30] [TRT] [I] Input filename:   /root/yolov5/runs/train/exp4/weights/best.onnx
[07/12/2022-09:29:30] [TRT] [I] ONNX IR version:  0.0.7
[07/12/2022-09:29:30] [TRT] [I] Opset version:    13
[07/12/2022-09:29:30] [TRT] [I] Producer name:    pytorch
[07/12/2022-09:29:30] [TRT] [I] Producer version: 1.11.0
[07/12/2022-09:29:30] [TRT] [I] Domain:           
[07/12/2022-09:29:30] [TRT] [I] Model version:    0
[07/12/2022-09:29:30] [TRT] [I] Doc string:       
[07/12/2022-09:29:30] [TRT] [I] ----------------------------------------------------------------
[07/12/2022-09:29:30] [TRT] [W] onnx2trt_utils.cpp:369: Your ONNX model has been generated with INT64 weights, while TensorRT does not natively support INT64. Attempting to cast down to INT32.
TensorRT: Network Description:
TensorRT:	input "images" with shape (1, 3, 640, 640) and dtype DataType.HALF
TensorRT:	output "output" with shape (1, 25200, 13) and dtype DataType.HALF
TensorRT: building FP16 engine in /root/yolov5/runs/train/exp4/weights/best.engine
[07/12/2022-09:29:31] [TRT] [I] [MemUsageChange] Init cuBLAS/cuBLASLt: CPU +83, GPU +164, now: CPU 3428, GPU 3452 (MiB)
[07/12/2022-09:29:31] [TRT] [I] [MemUsageChange] Init cuDNN: CPU +270, GPU +156, now: CPU 3698, GPU 3608 (MiB)
[07/12/2022-09:29:31] [TRT] [W] TensorRT was linked against cuDNN 8.4.1 but loaded cuDNN 8.0.5
[07/12/2022-09:29:31] [TRT] [I] Local timing cache in use. Profiling results in this builder pass will not be stored.
[07/12/2022-09:33:00] [TRT] [I] Some tactics do not have sufficient workspace memory to run. Increasing workspace size will enable more tactics, please check verbose output for requested sizes.
[07/12/2022-09:36:09] [TRT] [I] Detected 1 inputs and 4 output network tensors.
[07/12/2022-09:36:09] [TRT] [I] Total Host Persistent Memory: 156304
[07/12/2022-09:36:09] [TRT] [I] Total Device Persistent Memory: 775168
[07/12/2022-09:36:09] [TRT] [I] Total Scratch Memory: 0
[07/12/2022-09:36:09] [TRT] [I] [MemUsageStats] Peak memory usage of TRT CPU/GPU memory allocators: CPU 15 MiB, GPU 2186 MiB
[07/12/2022-09:36:09] [TRT] [I] [BlockAssignment] Algorithm ShiftNTopDown took 9.76303ms to assign 7 blocks to 101 nodes requiring 17152000 bytes.
[07/12/2022-09:36:09] [TRT] [I] Total Activation Memory: 17152000
[07/12/2022-09:36:09] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in building engine: CPU +1, GPU +14, now: CPU 1, GPU 14 (MiB)
[07/12/2022-09:36:09] [TRT] [W] The getMaxBatchSize() function should not be used with an engine built from a network created with NetworkDefinitionCreationFlag::kEXPLICIT_BATCH flag. This function will always return 1.
[07/12/2022-09:36:09] [TRT] [W] The getMaxBatchSize() function should not be used with an engine built from a network created with NetworkDefinitionCreationFlag::kEXPLICIT_BATCH flag. This function will always return 1.
TensorRT: export success, saved as /root/yolov5/runs/train/exp4/weights/best.engine (15.2 MB)

**Comment by glenn-jocher** [comment]
@huizhilei PyTorch inference and TensorRT inference use two completely different backends, I'm not sure anyone knows what to expect in terms of resource utilization other than simply acquiring empirical data, in which case the results are the results. 

If you see any bug or find room for improvement let us know.
